fs/grub/disk_io.c

```diff
@@ -128,30 +128,31 @@ int filemax;
 int
 rawread (int drive, int sector, int byte_offset, int byte_len, char *buf)
 {
+	/* FIXME: Simplify this function, as BootIdeReadSector() supports offset and buffering */
 	sector+=byte_offset/512;
 	byte_offset%=512;
 	
 	while(byte_len) {
-		int nThisTime=512;
-	       	if(byte_len<512) nThisTime=byte_len;
+		int nThisTime = (byte_len < 512 ? byte_len : 512);
```

Comment:
nit: Parentheses not necessary. If you add them, I'd instead suggest `[...] = (condition) ? true_value : false_value;`

Reply:
Improved.